src/utils.js

```diff
@@ -35,17 +38,43 @@ export function resolvePathFromService(services, name, parentName) {
 
 	let parentAddress = getServiceResolution(services, parentName);
 
-	let urlParts = (/^(([^:\/?#]+):)?(\/\/([^\/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?/g).exec(parentAddress);
+	let urlParts = getRegex().exec(parentAddress);
 
 	return urlParts[1] + urlParts[3] + join(
 		urlParts[5], '../', name
 	);
 }
 
+/**
+ * Given a resolved service url, generate a corresponding absolute path from the original service name.
+ * This is needed to resolve relative paths within the service name.
+ *
+ * @param {String} service The full service name requested to sofe, may include a relative path
+ * @param {String} url The resolved url of the service
+ *
+ * @return {String} A new url with. Possibly the same as the input url.
+ */
+export function getUrlFromService(service, url) {
+	let parts = getRegex().exec(service)[5].split('/');
+	parts = !parts[0] ? parts.slice(1) : parts;
+
+	const path = parts.slice(1).join('/');
```

Comment:
yeah, the `service` value is a qualified url when it is passed in. For example if you just do the following `import service from 'service!sofe';` then `service` will be `http://localhost/service`. So in the case above, we want to load the actual service, not a path to another file.

On the other hand if we `import dep from 'service/dep!sofe';` then `service` will be `http://localhost/service/dep.js`. In this case we want to get just `/dep.js`. The logic on the next line down tests if `path` is an empty string or not, and appends the path accordingly.